src/js/load-package.ts

```diff
@@ -502,6 +531,7 @@ export async function loadPackage(
     API.importlib.invalidate_caches();
   } finally {
     releaseLock();
+    return Array.from(loadedPackageData);
```

Comment:
Don't put returns inside finally blocks! It will discard exceptions! e.g., try out the following:
```js
function f() {
   try {
      throw new Error("hi");
   } finally {
      return 0;
   }
}
f()
```
This should go at the bottom of the try block.

Reply:
Let's also filter for the keys we want to expose:
```js
function filterPackageData({ name, version, file_name, package_type}: PrivatePackageData) : PackageData {
   return { name, version, file_name, package_type };
}
```
and then `return Array.from(loadedPackageData, filterPackageData);`.

Reply:
@ryanking13 maybe we should also change `file_name` and `package_type` to camelCase in the public packagedata? WDYT?

Reply:
> Don't put returns inside finally blocks! It will discard exceptions! e.g., try out the following:
> 
> ```js
> function f() {
>    try {
>       throw new Error("hi");
>    } finally {
>       return 0;
>    }
> }
> f()
> ```
> 
> The return should go at the bottom of the try block.

wow, I wasn't aware of such behavior. Thanks for letting me know 😮

Reply:
> @ryanking13 maybe we should also change file_name and package_type to camelCase in the public packagedata? WDYT?

Yeah, we haven't been very consistent about variable names, but it's probably a good idea to use camelCase for publicly exposed stuff.

`{ name, version, fileName, packageType }`